Design props for directional hysteresis

jafranc opened the pull request:
Design questions raised for implementing directional relative permeabilities and the number of directions

rrsettgast committed:
chore: Change GEOSX to GEOS in a bunch of places (#3219)

* GEOSX_USE to GEOS_USE

* GEOSX_ENABLE to GEOS_ENABLE

* GEOSX_LA to GEOS_LA

* GEOSX_TPL to GEOS_TPL

* GEOSX_ to GEOS_ excluding .py and .msub

* change version number

dkachuma committed:
refactor: Add component selector for fluid models (#3213)

* Add component selector

* Add test cases

* Fix integrated tests

* Remove thermal test

* Use integer sequence generator

* Use constexpr if

* thermal -> isThermalType

rrsettgast committed:
ci: Update CODEOWNERS (#3240)

* added multiple CODEOWNERS.

* exclusion for /src/schema and /BASELINE_NOTES.md

rrsettgast committed:
refactor: Towards modularization of physicsSolver (#3239)

* partial breakup of physics packages. Add cmake options for inclusion of each package

* move SourceFluxStatistics.xpp out of fieldSpecificaiton to physicsSolvers/fluidFlow

paveltomin committed:
feat: Check pore volume for all element types (#3233)

CusiniM committed:
refactor: fix creation of mass handling. (#3215)

Algiane committed:
docs: Fix wrong variable names in doc after PR #3219 (GEOSX_ -> GEOS_). (#3242)

Algiane committed:
fix: Fix link flags after pr#3219 (#3244)

* bug: fix wrong CMake variable name leading to missing linker flags.

* Fix double dependency error

Remove double link of statics libs when building some of the unit tests. This error creates 'multiple definition' error at link edition on Pangea3.

paveltomin committed:
changes to CODEOWNERS (#3238)

rrsettgast committed:
feat: Add partition information to terminal output (#3246)

* add function to output partition information

* add to CODEOWNERS file

rrsettgast committed:
change MPI_COMM_GEOSX to MPI_COMM_GEOS everywhere (#3259)

CusiniM committed:
chore: Update hypre (#3261)

rrsettgast committed:
fix: Fix segfault in table (#3268)

* change a range based loop that modifies the ranged container to a standard iterator loop. Isolate busted range bound in std::vector usage

sframba committed:
fix: robust point location for degenerate configurations (#3202)

* implemented method based on winding number

* using element global index for element choice in degenerate cases

* added global id criterion for points coincident with vertices, as well

* using base discretization for precomputing src/rcv

* enabled for high orders. Also removed extra high-order nodes where not needed
---------

Co-authored-by: Aurelien <aurelien.citrain@inria.fr>
Co-authored-by: acitrain <60715545+acitrain@users.noreply.github.com>

MelReyCG committed:
refactor: Remove circular dependencies from fileIO/common/codingutilities (#3247)

* DataTypes.hpp now only manage static types
- no runtime types nor Mpi constants
- DataTypes.cpp is not useful anymore.

* removing common -> fileIO

* removing common->codingUtilities

* logger has now a dedicated folder for upcoming files (future PRs)
---------

rrsettgast committed:
fix: testUnits failure on MacOS (#3269)

* change the SystemClock from std::chrono::system_clock to std::chrono::high_resolution_clock to avoid truncation problems on MacOS

MelReyCG committed:
feat: Transmissibility output (#3091)

* added transmissibility unit

* Added Cell-To-Cell output

* Add the logLevel viewkey & description (cherrypick from feature/rey/sourceflux-stats)

* Refactor the StencilOutput so the PackCollection can read it.
Point is to allow to do an HDF output, and remove the CSV one.

* Changing the Problem sub-groups init order so the outputs are at the end.
Point is to solve a bug where the StencilDataCollection buffer size are not known for the HDF chunk sizes.

* freeOnDevice fix test

rrsettgast committed:
fix: fix linking error on mac (#3270)

* Move the definition of fromDuration from Units.cpp to Units.hpp

CusiniM committed:
refactor: separate friction and hydraulic aperture - stress relation. (#3197)

CusiniM committed:
Update README.md (#3265)

CusiniM committed:
update LvArray submodule (#3277)

rrsettgast committed:
refactor: Remove "using namespace" in header files (#3255)

* remove using namespace dataRepository from headers files

* remove using namespace multifluid from NegativeTwoPhaseFlash.hpp

* removed using namespace constitutive from headers files

* removed using namespace fields from headers files

* removed using namespace stabilization from headers files

* removed using namespace fluxKernelsHelper from headers files

* remove using namespace surfaceGenerationKernelsHelpers from header files

rrsettgast committed:
docs: change GEOSX to GEOS in documentation (#3275)

* replace GEOSX with GEOS in conf.py file 

* update http address for opening new issues
---------

Co-authored-by: Matteo Cusini <matteo.cusini@gmail.com>

ryar9534 committed:
chore: Changing GEOSX to GEOS in external mesh import (#3278)

* Changing GEOSX to GEOS in external mesh import

* Update .integrated_tests.yaml

* Update BASELINE_NOTES.md

Oluwatobi-PM committed:
docs: stress_and_pressure_initialization (#3249)

* Updates to Gravity-Only Case

* User-defined Initialization Case

* Restructure examples into 2 folders

* update python scripts

* fix sphinx links

* update xml files and add smoke tests

* update rst file for gravity based model

* Clean Up Figures

---------

Co-authored-by: Oluwatobi Quadri Raji <oraji@sh02-ln01.stanford.edu>
Co-authored-by: Jian HUANG <jian.huang@totalenergies.com>
Co-authored-by: Herve Gross <40979822+herve-gross@users.noreply.github.com>
Co-authored-by: Oluwatobi Quadri Raji <oraji@sh02-ln03.stanford.edu>
Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

ryar9534 committed:
chore: add a few comments and small code fixes in addTransmissibilityCouplingPattern for poromechanics with conforming fractures (#3250)

---------

Co-authored-by: Ryan ARONSON (X) <j1067677@p4log01.main.glb.corp.local>

rrsettgast committed:
build: Remove remaining circular link dependencies (#3282)

* remove events from constitutive dependency list

* remove a bunch of redundant direct dependencies

* re-introduce options for dynamic libraries for each component

* move constitutive drivers outside of constitutive to resolve circular dependencies

* remove the use of blt_combine_static_libraries

* shared libs on by default in CI

* drop dependency of events on mesh. create temporary TriaxialDriver object in ~ProblemManager() to avoid constitutiveDrivers from being dropped by the linker

* add an outputDirectory in FunctionBase instead of grabbing from OutputBase. Also remove static data members in OutputBase. Also reenable TableFunction::print

* add WHOLE_ARCHIVE cmake decorations to linking for static libs. Lots of CMakeLists.txt cleanup

* cleanup unit test cmake files

* move constitutive driver documentation out on its own

jhuang2601 committed:
fix pythons script for a tutorial example (#3284)

paveltomin committed:
refactor: Reuse computeSinglePhaseFlux (#3283)

* Reuse computeSinglePhaseFlux

* Update SinglePhaseFVMKernels.hpp

dkachuma committed:
refactor: Use matrix-matrix interface in LAPACK solve (#3234)

* Use matrix interface

* uncrustify

* Remove unrelated changes

* Remove unncessesary assignment

---------

Co-authored-by: Matteo Cusini <49037133+CusiniM@users.noreply.github.com>

dkachuma committed:
feat: Use stability test in flash (#3206)

CusiniM committed:
feat: Add kernel callable dense LA solvers for small systems. (#3287)

CusiniM committed:
feat: add log at the end of the timestep with substeps info. (#3303)

rrsettgast committed:
add cmake install commands for component libraries (#3307)

rrsettgast committed:
fix: reenable floating point exceptions (#3300)

* reintroduce GEOS_ENABLE_FPE to GeosxOptions.cmake

* give Compositional[/Well]MultiphaseBase::m_maxRelativeCompDensChange a usable default value

* some modifications for fpe on testFlowStatistics

* update TPL paths. more fixes for CO2SolubilityDuanSun.cpp

* just disable FPE for testVTKImport

* allow failure of SeismicityRate_poromechanics_1d_smoke_01,SeismicityRate_poromechanics_1d_smoke_02,elas3D_Q3_abc_smoke_01,elas3D_Q3_abc_smoke_08,elas3D_Q3_abc_fs_smoke_01,elas3D_Q3_abc_fs_smoke_08 which all generate FPE's

dkachuma committed:
feat: Add immiscible water flash model (#3237)

* Add immiscible water flash model

Rclarson committed:
Making the number of directions contingent and able to handle either one or three directional input for relative permeability.  Needs work to be fully functional and design preferences appropriately implemented.

Rclarson committed:
partial revert "Making the number of directions contingent and able to handle either one or three directional input for relative permeability.  Needs work to be fully functional and design preferences appropriately implemented."

This reverts part of commit b665f9c9a0b025695f47cda74a1cf7d0886d0166.

Rclarson committed:
Getting rid of getFaceNormal changes

Rclarson committed:
changed a critical loop

herve-gross committed:
docs: Update COPYRIGHT (#3317)

* Correct error in COPYRIGHT statement
---------

Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

CusiniM committed:
docs: apply suggestions from joss review  (#3296)

* Modify build documents for problems from JOSS reviewers (#3258)

* add Apple MacOS build doc

* quick-start-template.cmake

* Update host-configs/quick-start-template.cmake

* add CODE_OF_CONDUCT and CONTRIBUTING file

* fix doxygen links in sphinx docs

* Update src/docs/sphinx/QuickStart.rst

* Doxygen for SolverBase

* Update Example.rst (#3315)

* add frontier scaling to documentation

* add the entire physicsSolver directory to the readthedocs doxygen

* add card on main documentation page for submitting issues

* fix a couple of typos

* Add instructions, postprocess script, and figures to WellboreECP problem

---------

Co-authored-by: Randolph Settgast <settgast1@llnl.gov>
Co-authored-by: Gerasimos Chourdakis <gerasimos.chourdakis@ipvs.uni-stuttgart.de>
Co-authored-by: Victor A. P. Magri <paludettomag1@llnl.gov>

jhuang2601 committed:
docs: update publications and fix a link (#3304)

* add a new publication on IJRMMS

* fix a link for triaxialDriver

sframba committed:
fix: Bug with elastic wave propagator at higher orders (#3294)

* fixed a bug due to PR 3202 concerning the elastic solver at higher orders

* added corresponding compute jacobian with corners

* added correction for ghost nodes

* Re-enable wave tests

---------

Co-authored-by: acitrain <60715545+acitrain@users.noreply.github.com>

paveltomin committed:
fix: Small fix (#3308)

* Fix bug in SinglePhasePoromechanicsConformingFractures.cpp
---------

Rclarson committed:
some changes to propogate variable numDir, will revert most

sytuannguyen committed:
feat: Temperature dependent single phase thermal conductivity (#3135)

jafranc commented on src/coreComponents/constitutive/relativePermeability/TableRelativePermeabilityHysteresis.cpp:
```diff
@@ -487,6 +500,8 @@ void TableRelativePermeabilityHysteresis::computeLandCoefficient()
   integer ipNonWetting = 0;
 
   integer const numPhases = m_phaseNames.size();
+  integer const numDir = m_drainageWettingNonWettingRelPermTableNames.size(0);
```
in order no to have to repeat every time the capture of `numDir`, maybe make is a member variable of the class and assign and check consistency in the `postInputInitialization()`

jafranc commented on src/coreComponents/constitutive/relativePermeability/TableRelativePermeabilityHysteresis.cpp:
```diff
@@ -166,7 +166,10 @@ void TableRelativePermeabilityHysteresis::postInputInitialization()
 
   using IPT = TableRelativePermeabilityHysteresis::ImbibitionPhasePairPhaseType;
 
+
```
Maybe assign here member variable `numDir` and check that it is consistent across all tables

Rclarson replied:
Yes, that could work.  Also need to add capability to use whichever input string is present out of optional inputs.

jafranc commented on src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp:
```diff
@@ -354,10 +357,23 @@ struct PPUPhaseFlux
-    auto const correction_term = LvArray::math::abs( faceNormal[0] + faceNormal[1] + faceNormal[2] + 1e-25 );
-    real64 const mobility = LvArray::math::abs( LvArray::tensorOps::AiBi< 3 >( phaseMob[er_up][esr_up][ei_up][ip], faceNormal ) )
-                            /correction_term;
+    integer const numDir = 3; // phaseMob[er_up][esr_up][ei_up][ip].size();    
```
this `numDir` could indeed be inferred from `phaseMob` number of elements.

jafranc commented on src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp:
```diff
@@ -503,17 +549,38 @@ struct C1PPUPhaseFlux
 
     // dC
+
+    if (numDir ==1)
```
As we are to that many if-branching, I would suggest investing time in refactoring into two functions and one and only `if (numDir==1)`

Rclarson replied:
I'll take a look at applying these changes

jafranc commented on src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp:
```diff
@@ -2043,23 +2045,29 @@ class DirichletFaceBasedAssemblyKernel : public FaceBasedAssemblyKernel< NUM_COM
       if( potDif >= 0 )              // the element is upstream
       {
-
+        if (numDir == 1)
```
this branch is inactive if l.2048 stays

jafranc commented on src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp:
```diff
@@ -147,7 +147,7 @@ class PhaseMobilityKernel : public isothermalCompositionalMultiphaseBaseKernels:
 
 
-    for( int dir = 0; dir < 3; ++dir )
+    for( int dir = 0; dir < 1; ++dir )
```
why is it forced to 1 here ?

jafranc reviewed (commented):
Few notes.
On general level, it would be easier to factored out the two cases `if(numDir==1)` and `if(numDir==3)` into functions and let the already defined functions to just do the dispatch.
It would be way easier to read and maintain.

Rclarson committed:
expanding the changes to all relative permeability models with some other fixes; task still needs to be completed

frankfeifan committed:
fix: Fix restart issue when running HydroFracture cases. (#3302)

dkachuma committed:
Add some initialisation to class members (#3323)

Rclarson committed:
change relperminterpolator (need to see if appropriate) and a couple other indexing that was incorrect and/or causing errors

Rclarson committed:
Revert "made jparams and kcurvparam directional input"

This reverts commit e1c148b9d235c8a1b6d5d1eb7f3a24c812f41537.

Conflicts:
	src/coreComponents/constitutive/relativePermeability/TableRelativePermeabilityHysteresis.cpp

CusiniM committed:
chore: update codeowners for integrated_tests.yaml (#3333)

Rclarson committed:
start of numDir based on inputs

jhuang2601 committed:
feat: print out fracture state for contact model (#3322)

Rclarson committed:
relpermbase minor change can revisit

Commit:
Revert "start of numDir based on inputs"

This reverts commit ecbb6acfe6796b0f4ef3500fae02c8d0d879056a.

rrsettgast committed:
docs: edits from joss review (#3332)

Rclarson committed:
fix the start of factoring out if statements

Commit:
Merge branch 'feature/larson/directional-hysteresis-design' of https://github.com/GEOS-DEV/GEOS into feature/larson/directional-hysteresis-design

rrsettgast committed:
update schema docs from previous PR's that forgot to generate the updates (#3341)

paveltomin committed:
chore: Update quick-start-template.cmake (#3326)

* Update quick-start-template.cmake

CusiniM committed:
chore: increase minimum required cmake version. (#3331)

* build: increase minimum required cmake version.

---------

Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

rrsettgast committed:
ci: Disable trigger of ci jobs when description is edited (#3346)

paveltomin committed:
fix: Check for zero permeability in J-function cap pressure (#3320)

* Update JFunctionCapillaryPressure.cpp

---------

Co-authored-by: Dickson Kachuma <81433670+dkachuma@users.noreply.github.com>
Co-authored-by: Pavel Tomin <“paveltomin@users.noreply.github.com”>
Co-authored-by: Matteo Cusini <49037133+CusiniM@users.noreply.github.com>

rrsettgast committed:
fix: Fix the sphinx errors present in the readthedocs log (#3343)

* turn on fail_on_warning in readthedocs.yml

* exclude blt and from sphinx log

* fix some sphinx errors from readthedocs log

* try using np.asarray to avoid the DepricationWarnings in several python scripts

CusiniM committed:
fix: fix rockylinux jobs and activate them. (#3336)

* fix: fix rockylinux jobs and activate them.

* clean up circular dependency between finiteVolume, fieldSpecification and linearAlgebra.

* use LvArray::NumericLimits

* try new certs procedure on rocky/rhel

* replace a+=b with a=a+b in wave solver to allow for fma operations

---------

Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

Algiane committed:
ci: add the possitility to use other build generators than ninja inside the ci (#3276)

* Add possibility to use other build generators than ninja in ci.

---------

Co-authored-by: Gaetan <gaetan.fuss.fr@gmail.com>
Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

CusiniM committed:
fix: modify and reactivate SeismicityRate  poroelastic integrated test. (#3318)

* Modify and reactivate SeismicityRate integratedTest.

* baseline notes.

* Update .integrated_tests.yaml

---------

Co-authored-by: Jian Huang <53012159+jhuang2601@users.noreply.github.com>
Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

rrsettgast committed:
docs: avoid offline documentation generation (#3350)

* Change formats:all to formats:pdf

cssherman committed:
docs: Generating sphinx datastructure tables on the fly (#3344)

* Generating sphinx datastructure tables on the fly

* Updating doc build instructions

* add cmake dependency of geosx_docs on geosx_build_datastructure_tables

* Using absolute paths for datastructure tables

---------

Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

Rclarson committed:
merge develop into branch for consistency with current tpls

Commit:
Merge branch 'feature/larson/directional-hysteresis-design' of https://github.com/GEOS-DEV/GEOS into feature/larson/directional-hysteresis-design

rrsettgast committed:
replace the word explosion with seismic charge (#3354)

Algiane committed:
ci: Time-saving version of PR #3159 that adds Pangea3 job with ppc emulation (#3340)

* Add pangea-3 job and emulation step.

* Replace relative path in cmake for P3-wave-solver host-config file.

* Build wave solver only in Pangea 3 job.

* Build Geos executable only in Pangea 3 job.

---------

Co-authored-by: Gaetan <159525405+Bubusch@users.noreply.github.com>

rrsettgast committed:
docs: Add scripts for user/commit stats and update CITATION.cff (#3357)

* scripts for user/commit stats

* modified repositoryStats_condeseDev.py to output authors in the citation.cff format

rrsettgast committed:
update DOI in CITATION.cff

matteofrigo5 committed:
feat: Augmented Lagrangian (slip and open modes)  (#3217)

* Implementation of slip and open modes for ALM

* Adding a new inputFile for ALM and bug (bubble functions) fixed

* Bug bubble functions gradient fixed

* Adding initial stress contribution for bubble functions - updating lambda functions

* Adding nested ALM

* Adding simultaneous and nested ALM 

* Moving the traction update into the friction model

* Moved contact constitutive behavior from ALM to the friction law

castelletto1 committed:
refactor: mixture density computation in multiphase poromechanics solvers (#3342)

* Simplifying body force computation

dkachuma committed:
fix: Remove shadow declarations (#3293)

* fix: Remove shadow declarations

* Rename variable

---------

Co-authored-by: Matteo Cusini <49037133+CusiniM@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump the github-actions group with 2 updates (#3324)

Bumps the github-actions group with 2 updates: [google-github-actions/auth](https://github.com/google-github-actions/auth) and [google-github-actions/setup-gcloud](https://github.com/google-github-actions/setup-gcloud).


Updates `google-github-actions/auth` from 2.1.3 to 2.1.5
- [Release notes](https://github.com/google-github-actions/auth/releases)
- [Changelog](https://github.com/google-github-actions/auth/blob/main/CHANGELOG.md)
- [Commits](https://github.com/google-github-actions/auth/compare/v2.1.3...v2.1.5)

Updates `google-github-actions/setup-gcloud` from 2.1.0 to 2.1.1
- [Release notes](https://github.com/google-github-actions/setup-gcloud/releases)
- [Changelog](https://github.com/google-github-actions/setup-gcloud/blob/main/CHANGELOG.md)
- [Commits](https://github.com/google-github-actions/setup-gcloud/compare/v2.1.0...v2.1.1)

---
updated-dependencies:
- dependency-name: google-github-actions/auth
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
- dependency-name: google-github-actions/setup-gcloud
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

victorapm committed:
feat: minimum interval for time-step increase after time-step cut (#3338)

* Add minDtIncreaseInterval and update time-stepping logic

* Use getter for minTimeStepIncreaseInterval

* change ats config for c1_ppy.ats...they appear invalid

* rebaseline

---------

Co-authored-by: Nicola Castelletto <38361926+castelletto1@users.noreply.github.com>
Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

rrsettgast committed:
build: update Mathpresso (#3360)

* update GEOS_TPL_TAG. Update version tag. enable mathpresso on osx. enable xml tools on osx

* runs on ruby with new LC path and new schema modifications from a previous contribution

* update LvArray hash. Move geosx_docs inside schema generation guard

paveltomin committed:
refactor: Trying to improve adaptive linear solver tolerance selection (#3163)

* improve adaptive linear solver tolerance selection. Added adaptiveExponent, adaptiveGamma, and strongestTol input.

* updated baselines for new fields

---------

Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

rrsettgast committed:
add code coverage label requirement (#3363)

sframba committed:
fix: testLifoStorage random fail on RockyLinux with clang+CUDA (#3362)

* added debug print for testLifoStorage

* force test for debugging purposes

* added extraction of TPL tag in ci

* corrected ci yml

* pb with yml

* indentation

* manually added tpl tag

* leftover dependency

* corrected tag extraction

* indentation

* building only failing test

* do not install

* do not run other tests

* commented print statements

* changed lifo test sizes for coherence

* changed lifo test sizes for coherence

* restoring yml files and updating testLifoStorage

* restored ci build script

Commit:
config for sherlock to merge

Commit:
Merge remote-tracking branch 'origin/develop' into feature/larson/directional-hysteresis-design

Conflicts:
	host-configs/Stanford/sherlock-gcc10-ompi4.1.2-openblas0.3.10.cmake
	src/coreComponents/LvArray

jafranc committed:
Merge branch 'feature/larson/directional-hysteresis' into feature/larson/directional-hysteresis-design

Commit:
retry

Commit:
Merge branch 'feature/larson/directional-hysteresis-design' of https://github.com/GEOS-DEV/GEOS into feature/larson/directional-hysteresis-design

Recon
